tests/h/accounts/views_test.py

```diff
@@ -59,6 +60,145 @@ def invalid_form(errors=None):
     return form
 
 
+class TestToXHRResponse(object):
+    """Unit tests for to_xhr_response()."""
+
+    def test_returns_given_result_if_not_xhr(self, pyramid_request):
+        """
+        If ``request`` isn't an XHR request it returns ``non_xhr_result``.
+
+        The calling view callable passes in the result that it would have
+        returned normally if this were not an XHR request as the
+        ``non_xhr_result`` argument. If the given ``request`` is not an XHR
+        request then ``non_xhr_result`` should just be returned unmodified.
+
+        """
+        pyramid_request.is_xhr = False
+        non_xhr_result = mock.Mock()
```

Comment:
In general, when you're just passing values around, it can often be quicker/clearer to use `mock.sentinel.SOMETHING` rather than constructing a `mock.Mock()` object. It has the advantage that the test will fail if you access an attribute on the object, making it clear that it's no longer being used as an opaque value.